src/stack.ts

```diff
@@ -294,3 +324,74 @@ function generateAppId(): string {
         .replace(/[^a-z0-9-.]/g, '-')
         .slice(-17);
 }
+
+/**
+ * Resources with this ID are hidden from humans
+ *
+ * They do not appear in the human-readable part of the logical ID
+ */
+const HIDDEN_FROM_HUMAN_ID = 'Resource';
+
+/**
+ * Resources with this ID are complete hidden from the logical ID calculation.
+ */
+const HIDDEN_ID = 'Default';
+const MAX_HUMAN_LEN = 240; // this is the value in CDK and seems like a good default to keep
+
+/**
+ * Calculates a unique ID for a set of textual components.
+ *
+ * This is forked from the internal cdk implementation with the removal of the hash suffix.
```

Comment:
Not a lawyer, but don't we have to retain the copyright information if we copy the code from the CDK repo?

It would be great if we could mark where we modified the code

Reply:
@flostadler I moved all this code to it's own file and added license information. Let me know what you think.